src/pages/write-smart-contracts/billboard-tutorial.md

```diff
@@ -0,0 +1,281 @@
+          (new-price (+ cur-price u10)))
+
+        ;; pay the contract
+        (unwrap! (stx-transfer? cur-price tx-sender (as-contract tx-sender)) (err ERR_STX_TRANSFER))
+
+        ;; update the billboard's message
+        (var-set billboard-message message)
+
+        ;; update the price
+        (var-set price new-price)
+
+        ;; return the updated price
+        (ok new-price)
+    )
+)
+```
+
+## Step 5: write a contract test
+
+At this point, the contract functions as intended, and can be deployed to the blockchain. However, it's good practice
+to write automated testing to ensure that the contract functions perform in the expected way. Testing can be valuable
+when adding complexity or new functions, as working tests can verify that any changes you make didn't fundamentally
+alter the way the functions behave.
+
+Open the `tests/billboard_test.ts` file in your IDE. In this step, you will add a single automated test to exercise the
+`set-message` and `get-message` functions of the contract.
+
+```ts
+import { Clarinet, Tx, Chain, Account, types } from 'https://deno.land/x/clarinet@v0.10.0/index.ts';
+import { assertEquals } from 'https://deno.land/std@0.90.0/testing/asserts.ts';
+
+Clarinet.test({
+  name: 'Ensure that the message can be set',
+  async fn(chain: Chain, accounts: Map<string, Account>) {
+    let wallet_1 = accounts.get('wallet_1')!;
+
+    let assetMaps = chain.getAssetsMaps();
+    const balance = assetMaps.assets['STX'][wallet_1.address];
+
+    let block = chain.mineBlock([
+      Tx.contractCall('billboard', 'set-message', [types.utf8('testing')], wallet_1.address),
+      Tx.contractCall('billboard', 'get-message', [], wallet_1.address),
+      Tx.contractCall('billboard', 'set-message', [types.utf8('testing...')], wallet_1.address),
+      Tx.contractCall('billboard', 'get-message', [], wallet_1.address),
+    ]);
+
+    assertEquals(block.receipts.length, 4);
+    assertEquals(block.height, 2);
+
+    block.receipts[1].result.expectUtf8('testing');
+
+    block.receipts[3].result.expectUtf8('testing...');
+
+    assetMaps = chain.getAssetsMaps();
+    assertEquals(assetMaps.assets['STX'][wallet_1.address], balance - 210);
+  },
+});
+```
+
+Modify the default imports in the boilerplate to include `{ Clarinet, Tx, Chain, Account, types }` from the `clarinet`
+Deno library.
+
+Give the test a descriptive name using the `name` field, then modify the test function to include the `chain` and
+`accounts` arguments. Make sure that you define a type for each of the arguments to satisfy the Typescript compiler.
+
+Using the Clarinet library, define variables to get a wallet address principal from the Clarinet configuration, and the
+balance of that address on the chain.
+
+The functional part of the test is defined using the `chain.mineBlock()` function, which simulates the mining of a
+block. Within that function, the test makes 4 contract calls (`Tx.contractCall()`), 2 calls to `set-message` and 2 calls
```

Comment:
a nit that we could review: `four contract calls [...], two calls [...] and two calls`